### PR TITLE
W-065

### PR DESCRIPTION
**Pull request by szymon-muszynski**
Added option to store basic and additional info on user's profile, connected to database.
Added 2 modes for each textbox - edit, read only mode, and save, where writing is allowed.

### CODE REVIEW

**Comment by WOJTASK1** [comment]
After changing view to another subpage and coming back to user profile, previously inserted avatar picture disappears. Please check this case.

File: backend/plortal/src/main/java/com/plortal/plortal/model/entity/User.java [hunk 1]
```diff
@@ -38,6 +38,14 @@ public class User {
     @Column(name = "role")
     private UserRole role;
 
+    @Size(max = 250)
+    @Column(name = "basicInfo")
+    private String basicInfo;
+
+    @Size(max = 250)
+    @Column(name = "additionalInfo")
```
Comment (wojciechwelfle) [inline backend/plortal/src/main/java/com/plortal/plortal/model/entity/User.java:46]: add length = 250

File: backend/plortal/src/main/java/com/plortal/plortal/model/entity/User.java [hunk 2]
```diff
@@ -38,6 +38,14 @@ public class User {
     @Column(name = "role")
     private UserRole role;
 
+    @Size(max = 250)
+    @Column(name = "basicInfo")
+    private String basicInfo;
```
Comment (wojciechwelfle) [inline backend/plortal/src/main/java/com/plortal/plortal/model/entity/User.java:43]: add length = 250

File: backend/plortal/src/main/java/com/plortal/plortal/controller/UserController.java [hunk 3]
```diff
@@ -103,4 +103,51 @@ public ResponseEntity<?> changePassword(@PathVariable Long userId, @RequestParam
         }
     }
 
+    @PostMapping("/update-basic-info/{userId}")
+    public ResponseEntity<?> updateBasicInfo(@PathVariable Long userId, @RequestParam String basicInfo) {
+        try {
```
Comment (wojciechwelfle) [inline backend/plortal/src/main/java/com/plortal/plortal/controller/UserController.java:108]: move methods to UserInfoController

File: backend/plortal/src/main/java/com/plortal/plortal/service/UserService.java [hunk 4]
```diff
@@ -19,4 +19,12 @@ public interface UserService {
     void deleteUser(Long id);
 
     void changePassword(Long userId, String oldPassword, String newPassword);
+
+    void updateBasicInfo(Long userId, String basicInfo);
+
+    void updateAdditionalInfo(Long userId, String additionalInfo);
+
+    String getBasicInfo(Long userId);
```
Comment (wojciechwelfle) [inline backend/plortal/src/main/java/com/plortal/plortal/service/UserService.java:27]: move to UserInfoService

File: backend/plortal/src/main/java/com/plortal/plortal/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -113,4 +113,46 @@ public void changePassword(Long userId, String oldPassword, String newPassword)
             throw new UserNotExistException();
         }
     }
+
+    @Override
+    public void updateBasicInfo(Long userId, String basicInfo) {
```
Comment (wojciechwelfle) [inline backend/plortal/src/main/java/com/plortal/plortal/service/impl/UserServiceImpl.java:118]: move to UserInfoServiceImpl

**Review by wojciechwelfle — CHANGES_REQUESTED** [review]
refactor todo

**Review by wojciechwelfle — APPROVED** [review]
lgtm

**Pull request merged**